--- madsonic_scripts/madsonic.sh.vanilla	2018-12-19 16:44:18.000000000 +0100
+++ madsonic_scripts/madsonic.sh	2018-12-20 20:23:30.000000000 +0100
@@ -9,18 +9,20 @@
 MADSONIC_HOME=/var/madsonic
 MADSONIC_HOST=0.0.0.0
 MADSONIC_PORT=4040
-MADSONIC_HTTPS_PORT=0
+MADSONIC_HTTPS_PORT=4443
 MADSONIC_CONTEXT_PATH=/
 MADSONIC_INIT_MEMORY=256
 MADSONIC_MAX_MEMORY=512
-MADSONIC_DEFAULT_MUSIC_FOLDER=/var/media/artists
-MADSONIC_DEFAULT_UPLOAD_FOLDER=/var/media/incoming
-MADSONIC_DEFAULT_PODCAST_FOLDER=/var/media/podcast
-MADSONIC_DEFAULT_PLAYLIST_IMPORT_FOLDER=/var/media/playlists/import
-MADSONIC_DEFAULT_PLAYLIST_EXPORT_FOLDER=/var/media/playlists/export
-MADSONIC_DEFAULT_PLAYLIST_BACKUP_FOLDER=/var/media/playlists/backup
-MADSONIC_DEFAULT_TRANSCODE_FOLDER=
-MADSONIC_DEFAULT_TIMEZONE=
+MADSONIC_DEFAULT_MUSIC_FOLDER=/media/music
+MADSONIC_DEFAULT_UPLOAD_FOLDER=/media/incoming
+MADSONIC_DEFAULT_PODCAST_FOLDER=/media/podcast
+MADSONIC_DEFAULT_PLAYLIST_IMPORT_FOLDER=/media/playlists/import
+MADSONIC_DEFAULT_PLAYLIST_EXPORT_FOLDER=/media/playlists/export
+MADSONIC_DEFAULT_PLAYLIST_BACKUP_FOLDER=/media/playlists/backup
+MADSONIC_DEFAULT_KEYSTORE_PASSWORD=madsonic
+MADSONIC_DEFAULT_KEYSTORE=${MADSONIC_HOME}/madsonic.keystore
+MADSONIC_DEFAULT_TRANSCODE_FOLDER=${MADSONIC_HOME}/transcode #/usr/local/bin	
+MADSONIC_DEFAULT_TIMEZONE=Europe/Berlin
 MADSONIC_PIDFILE=
 MADSONIC_UPDATE=true
 MADSONIC_GZIP=
@@ -29,6 +31,7 @@
 MADSONIC_PROXY_TYPE=http
 MADSONIC_PROXY_HOST=localhost
 MADSONIC_PROXY_PORT=8888
+MADSONIC_USE_SSL=true
 quiet=0
 
 usage() {
@@ -66,11 +69,16 @@
     echo "  --default-playlist-backup-folder=DIR  Configure Madsonic to use this folder for playlist backup."
     echo "                                        Default '/var/media/playlists/backup'"
     echo "  --default-transcode-folder=DIR        Configure Madsonic to use this folder for transcoder."
+    echo "  --default-keystore-folder=DIR         Configure Madsonic to use this keystore for HTTPS."
+    echo "                                        Default is ${MADSONIC_HOME}/madsonic.keystore."
+    echo "  --default-keystore-password=VALUE     Configure Madsonic to use this keystore password."
+    echo "                                        Default is 'madsonic'"
     echo "  --timezone=Zone/City                  Configure Madsonic to use other timezone for time correction"
     echo "                                        Example 'Europe/Vienna', 'US/Central', 'America/New_York'"
     echo "  --db=JDBC_URL                         Use alternate database. MySQL and HSQL are currently supported."
     echo "  --update=VALUE                        Configure Madsonic to look in folder /update for updates. Default 'true'"
     echo "  --gzip=VALUE                          Configure Madsonic to use Gzip compression. Default 'true'"
+    echo "  --use-ssl=VALUE                       Configure Madsonic to use TLS/SSL. Default 'true'."
     echo "  --quiet                               Don't print anything to standard out. Default false."
     exit 1 
 }
@@ -126,6 +134,12 @@
         --default-transcode-folder=?*)
             MADSONIC_DEFAULT_TRANSCODE_FOLDER=${1#--default-transcode-folder=}
             ;;
+        --default-keystore-folder=?*)
+           MADSONIC_DEFAULT_KEYSTORE=${1#--default-keystore-folder=}
+           ;;
+        --default-keystore-password=?*)
+          MADSONIC_DEFAULT_KEYSTORE_PASSWORD=${1#--default-keystore-password=}
+          ;;
         --timezone=?*)
            MADSONIC_DEFAULT_TIMEZONE=${1#--timezone=}
            ;;
@@ -138,6 +152,9 @@
         --db=?*)
             MADSONIC_DB=${1#--db=}
            ;;
+        --use-ssl=?*)
+          MADSONIC_USE_SSL=${1#--use-ssl=}
+          ;;
         --quiet)
             quiet=1
             ;;
@@ -165,7 +182,14 @@
     cd $(dirname $(readlink $0))
 fi
 
-${JAVA} -Xms${MADSONIC_INIT_MEMORY}m -Xmx${MADSONIC_MAX_MEMORY}m \
+SSL_PARAMETERS=""
+if [ $MADSONIC_USE_SSL = "true" ] && [ -e ${MADSONIC_DEFAULT_KEYSTORE} ]; then
+    SSL_PARAMETERS=" -Dmadsonic.ssl.keystore=${MADSONIC_DEFAULT_KEYSTORE} -Dmadsonic.ssl.password=${MADSONIC_DEFAULT_KEYSTORE_PASSWORD}"
+    echo "SSL/TLS enabled"
+    echo $SSL_PARAMETERS	
+fi
+
+${JAVA}${SSL_PARAMETERS} -Xms${MADSONIC_INIT_MEMORY}m -Xmx${MADSONIC_MAX_MEMORY}m \
   -Dmadsonic.home=${MADSONIC_HOME} \
   -Dmadsonic.host=${MADSONIC_HOST} \
   -Dmadsonic.port=${MADSONIC_PORT} \
